modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/ZonePartitionReplicaListener.java

```diff
@@ -122,9 +271,75 @@ public CompletableFuture<ReplicaResult> invoke(ReplicaRequest request, UUID send
         }
     }
 
+    private CompletableFuture<Void> processChangePeersAndLearnersReplicaRequest(ChangePeersAndLearnersAsyncReplicaRequest request) {
+        ZonePartitionId replicaGrpId = (ZonePartitionId) request.groupId().asReplicationGroupId();
+
+        RaftGroupService raftClient = raftCommandRunner instanceof RaftGroupService
```

Comment:
I understand that this is not your fault as you probably copied this code from somewhere, but it's very funny how bad we are at OOP =)